packages/blade/src/components/Switch/_decisions/decisions.md

```diff
@@ -0,0 +1,85 @@
+# Switch API Decisions <!-- omit in toc -->
+
+A switch component is used to quickly switch between two possible states. These are only used for binary actions that occur immediately after the user turn the switch on/off.
+
+<img src="./switch-thumbnail.png" width="380" alt="" />
+
+- [Design](#design)
+- [`Switch` API](#switch-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=31366-831710&t=qzTLJeIDluL4Ps23-0) to all variants of the Switch component
+
+
+## `Switch` API
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| isChecked | `boolean` | `false` | State of the switch, if explicitly set the switch will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components) |        |
+| defaultChecked | `boolean` | `false` | Default state of the switch, if isChecked is not provided the switch will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) |        |
+| onChange | `({ event, isChecked, value }) => void` | `undefined` | The function to be called when the switch state changes. |        |
+| isDisabled | `boolean` | `false` | Control whether the switch is disabled or not. |        |
+| accessibilityLabel | `string` | `undefined` | Provides accessible label for internal checkbox component that sets the aria-label prop for screen readers. |        |
```

Comment:
`accessibilityLabel` should be required right?

Reply:
No we can't make it optional. 

If people bind the label to the Switch, accessibilitlyLabel prop becomes redundant. 

```jsx
<label htmlFor="dark-mode">
   Dark mode
</label>
<Switch id="dark-mode" accessibilityLabel="Dark Mode" /> // Redundant aria-label, no need because label is binded to the switch already
  
// or

<label>
  Dark mode 
  // Redundant aria-label, no need because label is binded to the switch already
  <Switch accessibilityLabel="Dark Mode" />
</label>
```

Reply:
i don't even think anyone will care to use this prop 😮‍💨

Reply:
Hmm okay now that we are saying the "primary" function of the Switch won't be to used inside a form, but rather as a standalone button, then we can enforce accessibilityLabel, and treat `<label><swtich/></label>` as a rare case.